### PR TITLE
feat: auto-detect package manager

### LINKED ISSUE #1: Auto-detect the package manager being used
An attempt to run the install command using a package manager other than npm defaults to run types-sync using npm if not explicitly defined the package manager to be used in the `.types-syncrc.json`  file. 

In the case of yarn, it creates an extry `package-lock.json` file, while in the case of pnpm, it gets hung up.

undetect

It would be nice if the package manager used is auto-detected when the user does not contain the `.types-syncrc.json` file?

### PR DESCRIPTION
**Pull request by dhruwlalan**
Only If the `.types-syncrc.json` config file is not present or if the `"packageManager"` key on it is not defined, then the package manager will be auto-detected based on the lock file.

### CODE REVIEW

**Comment by maddhruv** [comment]
Hey, can you take a pull from the master, I have made a change to run CI on pull requests.

**Comment by maddhruv** [comment]
Also, it would be super helpful, if you can add this behavior in the `README` so as to let the users know that without passing the `packageManager` config, it can also detect the package manager

**Comment by dhruwlalan** [comment]
OK

File: README.md [hunk 1]
```diff
@@ -53,6 +53,9 @@ Add a `.types-syncrc.json` at the root of your project.
 | packageManager | `npm`/`yarn`/`pnpm`/`bolt` | package manager to use                  |
 | removeUnused   | `Boolean`                  | remove unsued types                     |
 
+Note: `types-sync` will automatically detect the package manager beign used. Explicitly Defining the
```
Comment (maddhruv) [inline README.md]:
```suggestion
Note: `types-sync` will automatically detect the package manager being used according to the lock files. Explicitly defining the
```

Typos

**Comment by dhruwlalan** [comment]
Let me know if the readme is correct or if you need any changes. :)

File: README.md [hunk 2]
```diff
@@ -53,6 +53,9 @@ Add a `.types-syncrc.json` at the root of your project.
...
+`packageManager` is optional and if defined will force `types-sync` to use that package manager.
```
Comment (maddhruv) [inline README.md]:
```suggestion
`packageManager` config option is optional and if defined will force `types-sync` to use the specified package manager.
```

**Review by maddhruv — CHANGES_REQUESTED** [review]
Please fix the typos and recommended changes in the readme

File: README.md (hunk 1 above)
Comment (maddhruv) [reply README.md]: Can you also make `Defining` lowercase?

**Comment by dhruwlalan** [comment]
done!

**Review by maddhruv — APPROVED** [review]
Thanks for your contribution 🎉 
tenor-250408114

**Comment by maddhruv** [comment]
Released `2.1.0` with the changes.
More info - https://github.com/maddhruv/types-sync/releases/tag/2.1.0

**Pull request merged**